name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TextUtilTest.java

```diff
@@ -191,4 +191,12 @@ public void testConcatenate()
         assertEquals("first-second", TextUtil.concatenate("first", "second", "-"));
     }
 
+    @Test
+    public void testCompareNullCheck()
+    {
+        assertEquals(-1, TextUtil.compare(null, "a"));
+        assertEquals(1, TextUtil.compare("a", null));
+        assertEquals(0, TextUtil.compare(null, null));
```

Comment:
```suggestion
        assertEquals(-1, TextUtil.compare(null, "a"));
        assertEquals(1, TextUtil.compare("a", null));
        assertEquals(0, TextUtil.compare(null, null));
        assertEquals(0, TextUtil.compare("abc", "abc"));
        assertTrue(TextUtil.compare("a", "b") < 0); // "a" should be less than "b"
        assertTrue(TextUtil.compare("z", "y") > 0); // "z" should be greater than "y"
        assertTrue(TextUtil.compare("abc", "ABC") != 0); // Depends on COLLATOR
        assertTrue(TextUtil.compare("café", "cafe") != 0); // Accents matter
```

Reply:
Interesting idea. I don't though about it before. Mabe for strings this is fine. But I think I want only check for ``null`` and let the underlaying comparer make the decision for not-null objects.
It's up to @buchen what to use.